public/app/features/alerting/unified/CloneRuleEditor.tsx

```diff
@@ -0,0 +1,94 @@
+import { cloneDeep } from 'lodash';
+import React from 'react';
+import { useAsync } from 'react-use';
+
+import { locationService } from '@grafana/runtime/src';
+import { Alert, LoadingPlaceholder } from '@grafana/ui/src';
+
+import { useDispatch } from '../../../types';
+import { RuleIdentifier, RuleWithLocation } from '../../../types/unified-alerting';
+import { RulerRuleDTO } from '../../../types/unified-alerting-dto';
+
+import { AlertRuleForm } from './components/rule-editor/AlertRuleForm';
+import { fetchEditableRuleAction } from './state/actions';
+import { rulerRuleToFormValues } from './utils/rule-form';
+import { isAlertingRulerRule, isGrafanaRulerRule, isRecordingRulerRule } from './utils/rules';
+import { createUrl } from './utils/url';
+
+export function CloneRuleEditor({ sourceRuleId }: { sourceRuleId: RuleIdentifier }) {
+  const dispatch = useDispatch();
+
+  const {
+    loading,
+    value: rule,
+    error,
+  } = useAsync(() => dispatch(fetchEditableRuleAction(sourceRuleId)).unwrap(), [sourceRuleId]);
+
+  if (loading) {
+    return <LoadingPlaceholder text="Loading the rule" />;
+  }
+
+  if (rule) {
+    const ruleClone = cloneDeep(rule);
+    changeRuleName(ruleClone.rule, generateCopiedRuleTitle(ruleClone));
+
+    return <AlertRuleForm prefill={rulerRuleToFormValues(ruleClone)} />;
+  }
+
+  if (error) {
+    return (
+      <Alert title="Error" severity="error">
+        {error.message}
+      </Alert>
+    );
+  }
+
+  return (
+    <Alert
+      title="Cannot duplicate. The rule does not exist"
+      buttonContent="Go back to alert list"
+      onRemove={() => locationService.replace(createUrl('/alerting/list'))}
+    />
+  );
+}
+
+function generateCopiedRuleTitle(originRuleWithLocation: RuleWithLocation): string {
```

Comment:
🙌

Reply:
@konrad147 , would you be able to confirm whether the newly duplicate alert rule will also produce an editable copy? (instead of a provisioned / GUI-editing limited copy).

Once manually-tailored alert rules are modified through the provisioning API they are forever locked from any Web GUI editing. Perhaps this PR would just create a copy of the same nature but I just wanted to double-check.

Reply:
That's a good question. This PR allows only duplicating non-provisioned alert rules. I'll check if we can offer copying to provisioned ones.
Regarding editing provisioned rules, there is another PR that allows rules created by the API to be editable https://github.com/grafana/grafana/pull/58410

Reply:
I updated the PR. Cloning of provisioned rules will be possible (but the cloned rule will not be regarded as provisioned one)